src/leader.py

```diff
@@ -152,6 +163,62 @@ def lead(self):
             self.turn(code, position)
 
 
+    def demo_circle(self):
+        """
+        Controls the leader vehicle in a circle for demo purposes.
+
+        Drives the leader vehicle in a continuous circle till script is
+        terminated.
+        """
+        while True:
+            self.fw.turn(DEMO_LEFT_TURN)
+            self.bw.speed(DEMO_SPEED)
+            self.bw.forward()
+
+
+    def demo_line(self):
+        """
+        Controls the leader vehicle in a line for demo purposes.
+
+        Drives the leader vehicle in a line then turns around and repeats till
+        script is terminated.
+        """
+        while True:
+
+            straight_timer = time.time()
```

Comment:
You might have to find a way to do this without a timer since it will keep resetting every cycle since you will probably have to take out all of the while loops